src/DataLakeStore/DataLakeStore/DataPlaneModels/DataLakeStoreFileSystemClient.cs

```diff
@@ -777,49 +777,24 @@ public void GetFileProperties(string accountName, string path, bool getAclUsage,
         /// <param name="filter">Query to match items in trash</param>
         /// <param name="count">Minimum number of entries to search for</param>
         /// <param name="cmdletCancellationToken">CancellationToken</param>
-        public IEnumerable<TrashEntry> EnumerateDeletedItems(string accountName, string filter, int count, CancellationToken cmdletCancellationToken = default(CancellationToken))
+        public IEnumerable<TrashEntry> EnumerateDeletedItems(string accountName, string filter, int count, Cmdlet cmdlet, CancellationToken cmdletCancellationToken = default(CancellationToken))
         {
-            IEnumerable<TrashEntry> result = null;
-            Task enumerateTask = null;
-            bool completed = false;
-            object syncPrimitive = new object();
-
-            var progressTracker = new Progress<EnumerateDeletedItemsProgress>();
-            progressTracker.ProgressChanged += (s, e) =>
-            {
-                // TODO: Update job progress here
-                lock (syncPrimitive)
-                {
-                    if (e.NumFound >= count || String.IsNullOrEmpty(e.NextListAfter))
-                    {
-                        completed = true;
-                    }
-
-                    Monitor.Pulse(syncPrimitive);
-                }
-            };
-
-            enumerateTask = Task.Run(() =>
+            var client = AdlsClientFactory.GetAdlsClient(accountName, _context);
+            if (_isDebugEnabled)
             {
-                result = AdlsClientFactory.GetAdlsClient(accountName, _context).EnumerateDeletedItems(filter, "", count, progressTracker, cmdletCancellationToken);
-            }, cmdletCancellationToken);
-
-            // Keep printing (?) the updates returned by successive calls by the SDK to the backend, until the whole query completes
-            while(true)
+                IEnumerable<TrashEntry> result = null;
+                // Api call below consists of multiple rest calls so multiple debug statements will be posted
+                // so since we want to the debug lines to be updated while the command runs, we have to flush the debug statements in queue and thats why we want to do it this way
+                var enumerateTask = Task.Run(() => {
+                result = client.EnumerateDeletedItems(filter, "", count, null, cmdletCancellationToken);
+                }, cmdletCancellationToken);
+                TrackTaskProgress(enumerateTask, cmdlet, null, cmdletCancellationToken);
```

Comment:
@rahuldutta90 what does the `TrackTaskProcess` method do differently than the `while` loop that was previously being used?

Reply:
@cormacpayne  TrackTaskProcess will exit if the task throws a exception, where as the while loop depends on "completed" to exit and "completed" will be true when "e.NumFound >= count || String.IsNullOrEmpty(e.NextListAfter)". Now imagine customer does not have permissions to run this operation and they will get a remote exception from ADLS, so "client.EnumerateDeletedItems" will throw an exception, however progresschanged will never be triggered and "completed" wil remain false.

TrackTaskProcess checks the status of task itself.